test/metabase/query_processor/middleware/update_used_cards_test.clj

```diff
@@ -108,4 +109,16 @@
         (is (= now
                (-> (t2/select-one-fn :last_used_at :model/Card card-id-2)
                    t/offset-date-time
-                   (.withNano 0))))))))
+                   (.withNano 0))))))
+    (testing "if metabase is in read only mode still set the last_used_at timestamp"
+      (mt/with-temp
+        [:model/Card {card-id-1 :id} {:last_used_at one-hour-ago}]
+        (try
+          (cloud-migration/read-only-mode! true)
```

Comment:
We have a nice utility function for this, `(mt/with-temporary-setting-values [read-only-mode true] ...)`, that makes this a little cleaner (and makes it harder to forget to reset the setting when you're done testing 😅).

Reply:
Thanks I will swap for that. I had just looked at how it was tested in the cloud-migration tests, so I'll update those too.